From 40a7174086b12445a57f0ee16c5117b0d7e6cced Mon Sep 17 00:00:00 2001
From: Alejandro Martinez Ruiz <alex@nowcomputing.com>
Date: Tue, 29 Jan 2013 17:17:51 +0100
Subject: [PATCH] Revert "f2fs: add un/freeze_fs into super_operations"

The (un)freeze mechanism was changed in 5accdf82ba25cacefd6c1867f1704beb4d244cdd
upstream. We just drop support for (un)freezing.

Signed-off-by: Alejandro Martinez Ruiz <alex@nowcomputing.com>
---
 fs/f2fs/gc.c    |  5 -----
 fs/f2fs/inode.c |  3 ---
 fs/f2fs/super.c | 18 ------------------
 3 files changed, 26 deletions(-)

diff --git a/fs/f2fs/gc.c b/fs/f2fs/gc.c
index 1496159..22aee03 100644
--- a/fs/f2fs/gc.c
+++ b/fs/f2fs/gc.c
@@ -44,11 +44,6 @@ static int gc_thread_func(void *data)
 		if (kthread_should_stop())
 			break;
 
-		if (sbi->sb->s_writers.frozen >= SB_FREEZE_WRITE) {
-			wait_ms = GC_THREAD_MAX_SLEEP_TIME;
-			continue;
-		}
-
 		/*
 		 * [GC triggering condition]
 		 * 0. GC is not conducted currently.
diff --git a/fs/f2fs/inode.c b/fs/f2fs/inode.c
index 91ac7f9..a49f696 100644
--- a/fs/f2fs/inode.c
+++ b/fs/f2fs/inode.c
@@ -258,7 +258,6 @@ void f2fs_evict_inode(struct inode *inode)
 	if (inode->i_nlink || is_bad_inode(inode))
 		goto no_delete;
 
-	sb_start_intwrite(inode->i_sb);
 	set_inode_flag(F2FS_I(inode), FI_NO_ALLOC);
 	i_size_write(inode, 0);
 
@@ -268,8 +267,6 @@ void f2fs_evict_inode(struct inode *inode)
 	ilock = mutex_lock_op(sbi);
 	remove_inode_page(inode);
 	mutex_unlock_op(sbi, ilock);
-
-	sb_end_intwrite(inode->i_sb);
 no_delete:
 	clear_inode(inode);
 }
diff --git a/fs/f2fs/super.c b/fs/f2fs/super.c
index 8555f7d..a5595e7 100644
--- a/fs/f2fs/super.c
+++ b/fs/f2fs/super.c
@@ -166,22 +166,6 @@ int f2fs_sync_fs(struct super_block *sb, int sync)
 	return 0;
 }
 
-static int f2fs_freeze(struct super_block *sb)
-{
-	int err;
-
-	if (sb->s_flags & MS_RDONLY)
-		return 0;
-
-	err = f2fs_sync_fs(sb, 1);
-	return err;
-}
-
-static int f2fs_unfreeze(struct super_block *sb)
-{
-	return 0;
-}
-
 static int f2fs_statfs(struct dentry *dentry, struct kstatfs *buf)
 {
 	struct super_block *sb = dentry->d_sb;
@@ -253,8 +237,6 @@ static struct super_operations f2fs_sops = {
 	.evict_inode	= f2fs_evict_inode,
 	.put_super	= f2fs_put_super,
 	.sync_fs	= f2fs_sync_fs,
-	.freeze_fs	= f2fs_freeze,
-	.unfreeze_fs	= f2fs_unfreeze,
 	.statfs		= f2fs_statfs,
 };
 
-- 
1.8.2.1.422.gdcd8c09

